src/app/locator/qgsinbuiltlocatorfilters.cpp

```diff
@@ -163,3 +164,80 @@ void QgsActionLocatorFilter::searchActions( const QString &string, QWidget *pare
     }
   }
 }
+
+QgsActiveLayerFeaturesLocatorFilter::QgsActiveLayerFeaturesLocatorFilter( QObject *parent )
+  : QgsLocatorFilter( parent )
+{
+  setUseWithoutPrefix( false );
+}
+
+void QgsActiveLayerFeaturesLocatorFilter::fetchResults( const QString &string, const QgsLocatorContext &, QgsFeedback *feedback )
+{
+  if ( string.length() < 3 )
+    return;
+
+  QgsVectorLayer *layer = qobject_cast< QgsVectorLayer *>( QgisApp::instance()->activeLayer() );
+  if ( !layer )
+    return;
+
+  int i = 0;
+  int found = 0;
+  QgsExpression dispExpression( layer->displayExpression() );
+  QgsExpressionContext context;
+  context.appendScopes( QgsExpressionContextUtils::globalProjectLayerScopes( layer ) );
+  dispExpression.prepare( &context );
+
+  Q_FOREACH ( const QgsField &field, layer->fields() )
+  {
+    if ( feedback->isCanceled() )
+      return;
+
+    QString exp = QStringLiteral( "%1 ILIKE '%%2%'" ).arg( QgsExpression::quotedColumnRef( field.name() ),
+                  string );
```

Comment:
I think it's fine - at least, any fix belongs in the individual expression compiler's quotedValue methods, not here.